feat(filter): Add less/sass/test page for filter component

jeff-phillips-18 opened the pull request:
## Description
Adds necessary selectors for filter components.

## Link to rawgit and/or image
http://rawgit.com/jeff-phillips-18/patternfly/filter-dist/dist/tests/filter.html

## PR checklist (if relevant)

- [ ] **Cross browser**: works in IE9
- [ ] **Cross browser**: works in IE10
- [ ] **Cross browser**: works in IE11
- [ ] **Cross browser**: works in Edge
- [x] **Cross browser**: works in Chrome
- [x] **Cross browser**: works in Firefox
- [x] **Cross browser**: works in Safari
- [ ] **Cross browser**: works in Opera
- [x] **Responsive**: works in extra small, small, medium and large view ports.
- [x] **Preview the PR**: An image or a URL for designer to preview this PR is provided.

andresgalante commented on src/less/filter.less:
```diff
@@ -0,0 +1,44 @@
+.filter-pf {
+  .filter-pf-category-select {
```
Let's avoid nesting this selectors, it'll just increase specificty

jeff-phillips-18 replied:
Fixed.

cliffpyles commented:
Other than the nested selector mentioned above, it looks good to me. 👍

jeff-phillips-18 commented:
Sorry, my dist branch was built incorrectly. Should be accurate now.

andresgalante commented on src/less/filter.less:
```diff
@@ -0,0 +1,46 @@
+  padding: 5px 0 6px 5px;
+}
+.filter-pf-fields .form-group {
+  width: 275px;
```
The width of the component should be regulated by it's parent container, I don't think we should not define a fix width.

andresgalante commented on src/less/filter.less:
```diff
@@ -0,0 +1,46 @@
+}
+.filter-pf-select {
+  .dropdown.btn-group {
+    width: 100%;
```
Are you sure we want to extend this element to the width of it's container? honestly I don't know :)

jeff-phillips-18 replied:
You are correct, these are not necessary.

andresgalante commented on src/less/filter.less:
```diff
@@ -0,0 +1,46 @@
+    min-width: 100%;
+  }
+  .caret {
+    margin-top: -5px;
```
the hack for vertical centering is not a negative margin top but a `transform: translateY(-50%);`

andresgalante commented on src/less/filter.less:
```diff
@@ -0,0 +1,46 @@
+  background-image: none;
+  color: @color-pf-black-500;
+  font-size: 12px;
+  font-style: italic;
```
This `italic` is making the caret look weird.

andresgalante commented on src/less/filter.less:
```diff
@@ -0,0 +1,46 @@
+    right: 12px;
+  }
+}
+.filter-pf-select-dropdown {
```
I am ok with all this font treatment as long as we make sure that it's consistent with other placeholders for inputs.

andresgalante commented on src/sass/converted/patternfly/_filter.scss:
```diff
@@ -0,0 +1,46 @@
+  width: 100%;
+}
+
+.filter-pf-active-label {
```
Where is this label in the HTML?

jeff-phillips-18 replied:
This is another item on the toolbar showing current filters, this is not shown in the test page

andresgalante commented on src/less/filter.less:
```diff
@@ -0,0 +1,46 @@
+  font-weight: 400;
+  padding-right: 25px;
+  text-align: left;
+  width: 100%;
```
if you remove the other 100%, maybe remove this one too

jeff-phillips-18 replied:
done

andresgalante commented:
@jeff-phillips-18 I left you a bunch of comments, I hope they make sense 😄

jeff-phillips-18 commented on src/less/filter.less:
```diff
@@ -0,0 +1,46 @@
+    margin-top: -5px;
+    position: absolute;
+    top: 50%;
+    right: 12px;
```
10px seems to work fine.

jeff-phillips-18 commented:
Updated per comments. Thanks @andresgalante

jeff-phillips-18 commented:
Regression test failures do not happen locally, only in travis.

```
$ npm run regressions pagination-table-view

> patternfly@0.0.0-semantically-released regressions /Users/jeffreyphillips/repositories/patternfly
> node ./backstop/test "pagination-table-view"

Config detected:
COMMAND | Executing core for `test`
createBitmaps | Selected 1 of 1 scenarios.
Starting Chromy: {"chromeFlags":["--disable-gpu","--force-device-scale-factor=1","--disable-infobars=true","--window-size=1280,720"],"port":9222,"waitTimeout":30000,"visible":false}
9222 Chrome v63 detected.
9222 INFO >  BackstopTools have been installed.
      COMMAND | Executing core for `report`
      compare | OK: pagination-table-view backstop_default_pagination-table-view_0_document_0_large-device.png
       report | Test completed...
       report | 1 Passed
       report | 0 Failed
       report | Writing browser report
       report | Browser reported copied
       report | Copied configuration to: /Users/jeffreyphillips/repositories/patternfly/backstop_data/html_report/config.js
      COMMAND | Executing core for `openReport`
   openReport | Opening report.
      COMMAND | Command `report` successfully executed in [0.762s]
      COMMAND | Command `test` successfully executed in [4.624s]
tests completed without failures



$ npm run regressions table-view-navbar

> patternfly@0.0.0-semantically-released regressions /Users/jeffreyphillips/repositories/patternfly
> node ./backstop/test "table-view-navbar"

Config detected:
COMMAND | Executing core for `test`
createBitmaps | Selected 1 of 1 scenarios.
Starting Chromy: {"chromeFlags":["--disable-gpu","--force-device-scale-factor=1","--disable-infobars=true","--window-size=1280,720"],"port":9222,"waitTimeout":30000,"visible":false}
9222 Chrome v63 detected.
9222 INFO >  BackstopTools have been installed.
      COMMAND | Executing core for `report`
      compare | OK: table-view-navbar backstop_default_table-view-navbar_0_document_0_large-device.png
       report | Test completed...
       report | 1 Passed
       report | 0 Failed
       report | Writing browser report
       report | Browser reported copied
       report | Copied configuration to: /Users/jeffreyphillips/repositories/patternfly/backstop_data/html_report/config.js
      COMMAND | Executing core for `openReport`
   openReport | Opening report.
      COMMAND | Command `report` successfully executed in [0.467s]
      COMMAND | Command `test` successfully executed in [3.812s]
tests completed without failures
```

jeff-phillips-18 commented:
The test should not have worked locally. I have fixed and updated the changed files. Changes were do to removing the 25% width on filter components in a toolbar.
